#152220230 Implement user signup

tomipaul opened the pull request:
#### What does this PR do?
Implement user signup

#### Description of Task to be completed?
- Create utility functions, controllers and middleware
- Define route and mount controllers/middleware

#### What are the relevant pivotal tracker stories?
#152220230

#### Screenshots (if appropriate)
nil

tomipaul committed:
chore(server): update migrations

tomipaul committed:
chore(server): update models

tomipaul committed:
chore(server): add configuration for sequelize-cli

tomipaul committed:
chore(server): turn off operatorsAliases

use symbol operators not strings, this is important for security considerations

tomipaul committed:
feat(server/utils) add utility functions

generate and verify token
validate signup payload
extract id, username and email from a User model instance
get a user

tomipaul committed:
feat(server/controllers): implement user authentication

add create user method
add authenticate user method

tomipaul committed:
feat(server/controllers): implement error handling middleware

tomipaul committed:
feat(server/routes): add route for user signup

tomipaul committed:
chore: update project dependencies and scripts

tomipaul committed:
refactor(server): separate mounting of handlers and middlewares

tomipaul committed:
feat(server): mount middlewares, routers, and handlers

tomipaul committed:
test: update tests

tomipaul commented:
@supermike, @3m3kalionel @ayodelevm please review

mikey2020 commented on src/server/controllers/AuthController.js:
```diff
@@ -0,0 +1,84 @@
+                return generateToken(user, rsaKey);
+              }
+              const error = new Error('Invalid Password');
+              error.code = 422;
```
Am curious to why you decided to return a `422` when you have an invalid password

mikey2020 replied:
You  used `err` before but now you are using `error`, you should try and be consistent when defining variables

tomipaul committed:
fix: implement review feedback

change err to error in all files
change error code from 422 to 401 for invalid password

tomipaul commented:
@mikey2020 please affirm that the suggested corrections have been made
